test/Interop/Cxx/class/conforms-to.swift

```diff
@@ -1,10 +1,10 @@
 // RUN: %empty-directory(%t)
 // RUN: %target-swift-frontend %S/Inputs/conforms-to-imported.swift -module-name ImportedModule -emit-module -emit-module-path %t/ImportedModule.swiftmodule
 
-// RUN: %target-typecheck-verify-swift -verify-ignore-unknown -I %t -I %S/Inputs -module-name SwiftTest -enable-experimental-cxx-interop
-// RUN: %target-typecheck-verify-swift -verify-ignore-unknown -I %t -I %S/Inputs -module-name SwiftTest -cxx-interoperability-mode=swift-5.9
-// RUN: %target-typecheck-verify-swift -verify-ignore-unknown -I %t -I %S/Inputs -module-name SwiftTest -cxx-interoperability-mode=swift-6
-// RUN: %target-typecheck-verify-swift -verify-ignore-unknown -I %t -I %S/Inputs -module-name SwiftTest -cxx-interoperability-mode=upcoming-swift
+// RUN: %target-typecheck-verify-swift -verify-ignore-unknown -I %t -I %S/Inputs -module-name SwiftTest -enable-experimental-cxx-interop -disable-availability-checking
```

Comment:
I was getting errors like these in this test case after my changes:
error: unexpected note produced: `add @available attribute to enclosing global function` for these functions:
```
func caller(_ x: DerivedFromHasPlay)
func caller(_ x: DerivedFromDerivedFromHasPlay)
func caller(_ x: HasTestAndPlay)
func caller(_ x: DerivedFromHasTestAndPlay)

```
So I thought the easiest way to suppress these errors would be to disable availability checking for this test case. 

But @egorzhdan do you think this could this be indicative of some other unrelated issue? All these types are actually deriving from a base immortal type (a kind of "import_reference" type) which might be impacted by this patch doing changes to FRT inference in cases of c++ inheritance.